src/Tuura/Plato/Translation.hs

```diff
@@ -76,8 +78,30 @@ validateInterface signs circuit
   where
     unused       = filter ((==Unused) . interface circuit) signs
 
-cartesianProduct :: NonEmpty.NonEmpty [a] -> [[a]]
-cartesianProduct l = sequence (NonEmpty.toList l)
+cartesianProduct :: Ord a => NonEmpty.NonEmpty [a] -> [[a]]
+cartesianProduct l = removeSupersets sortByLength
+  where
+    sequenced    = sequence (NonEmpty.toList l)
+    removeDupes  = map nub sequenced
+    removeNull   = filter (not . null) removeDupes
+    sortAllLists = map sort removeNull
+    sortByLength = sortBy (comparing length) sortAllLists
+
+removeSupersets :: Eq a => [[a]] -> [[a]]
+removeSupersets s = filter (not . null) result
+  where
+    prev n  = take n s
+    check x = checkForSupersets (s!!x) (prev x)
+    result  = map (\x -> if check x
+                        then []
+                        else s!!x
+                 )[0..(length s) - 1]
+
+checkForSupersets :: Eq a => [a] -> [[a]] -> Bool
+checkForSupersets current previous = any (`elem` subs) previous
```

Comment:
That's fair. I think I just misunderstood the implementation of `isSubsequenceOf`. I just assumed it would run `subsequences` each time, then check all of these with something like `elem`. I thought if I made sure it only ran `subsequences` once, then it would be quicker. 

Running your example (with `current = [1..200]`) proves your method works almost instantaneously, whereas mine runs for a while, makes my computer very warm, and then the process kills itself after a while, so yours is most definitely better :)